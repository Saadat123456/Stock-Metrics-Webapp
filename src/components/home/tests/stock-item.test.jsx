import { render, screen } from '@testing-library/react';
import { createMemoryHistory } from 'history';
import { BrowserRouter as Router } from 'react-router-dom';
import React from 'react';
import mockData from '../../../redux/StockData/mock_data';
import StockItem from '../stock-item';

it('renders correctly', () => {
  const history = createMemoryHistory();
  const { container } = render(
    <Router history={history}>
      <StockItem stock={mockData[0]} />
    </Router>,
  );
  expect(container.firstChild).toMatchSnapshot();
  const stockItemElement = screen.getByText(/Bitcoin/i);
  expect(stockItemElement).toBeInTheDocument();
});
