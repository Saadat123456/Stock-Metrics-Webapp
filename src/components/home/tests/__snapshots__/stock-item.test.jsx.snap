// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<a
  class="list-item-link position-1"
  data-testid="item-1"
  href="/details/1"
  id="item-1"
>
  <div
    class="next-btn"
  >
    <svg
      fill="currentColor"
      height="24"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M11.999 1.993c-5.514.001-10 4.487-10 10.001s4.486 10 10.001 10c5.513 0 9.999-4.486 10-10 0-5.514-4.486-10-10.001-10.001zM12 19.994c-4.412 0-8.001-3.589-8.001-8s3.589-8 8-8.001C16.411 3.994 20 7.583 20 11.994c-.001 4.411-3.59 8-8 8z"
      />
      <path
        d="M12 10.994H8v2h4V16l4.005-4.005L12 7.991z"
      />
    </svg>
  </div>
  <li
    class="list-item"
  >
    <div
      class="price"
    >
      <div>
        41152.271
         $
      </div>
    </div>
    <div
      class="representation"
    >
      <span>
        Bitcoin
      </span>
      <span>
        ( 
        BTC
         )
        <svg
          class="star"
          version="1.1"
          viewBox="0 0 194.22 184.73"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            transform="translate(-262.89 -462.86)"
          >
            <g
              transform="translate(10.175 71.896)"
            >
              <path
                d="M369.21 535.67l-27.378-13.792-26.883 14.735 4.657-30.3-22.321-21.014 30.256-4.935 13.088-27.722 14.043 27.251 30.409 3.881-21.577 21.776z"
                fill="#fc0"
                transform="rotate(1.005 35281.323 -22271.698) scale(2.21184)"
              />
              <path
                d="M380.03 451.38l-30.72 41.819 97.411-31.899z"
                fill="#ffe680"
              />
              <path
                d="M349.81 492.31v51.942l-60.63 31.117z"
                fill="#fd5"
              />
              <path
                d="M349.81 492.31l59.613 81.476-11.189-65.984z"
                fill="#fd5"
              />
              <path
                d="M253.1 461.37l96.715 30.942-29.882-40.842z"
                fill="#ffe680"
              />
              <path
                d="M349.81 492.31V391.09l29.83 60.357z"
                fill="#fd5"
              />
              <path
                d="M289.8 574.53l11.261-66.541 48.757-15.679z"
                fill="#ffd42a"
              />
            </g>
          </g>
        </svg>
      </span>
    </div>
    <div
      class="supply"
    >
      <div
        class="tooltip"
        data-title="Total Supply"
      >
        TS: 
      </div>
      19010687
    </div>
  </li>
</a>
`;
